File: solutions/c/pangram/2/pangram.c
#include "pangram.h"

bool is_pangram(const char *sentence) {
    if (!sentence) {
        return false;
    }
    size_t checksum = 0;
    for (size_t i = 0; i < strlen(sentence); i++) {
        if (isalpha(sentence[i])) {        // to avoid not letters
            char c = tolower(sentence[i]); // A -> a
            checksum |= 1 << (c - 'a');
        }
    }
    return checksum == 0x3ffffff;
}

File: solutions/go/pangram/2/pangram.go
package pangram

import (
    "unicode"
)

const checkPangram = 0x03FFFFFF 

func IsPangram(input string) bool {
    count := 0
	for _, letter := range input {
        letter = unicode.ToLower(letter)
        if letter >= 'a' && letter <= 'z' {
            count |= 1 << (letter - 'a');
        }
    }
    return count == checkPangram
}
